src/Cake.NuGet/Install/NuGetPackageInstaller.cs

```diff
@@ -115,6 +131,22 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
             return project.GetFiles(path, package, type);
         }
 
+        private static DirectoryPath GetPackagePath(IDirectory root, string package)
+        {
+            var directories = root.GetDirectories("*", SearchScope.Current).ToArray();
+            return directories.FirstOrDefault(p => p.Path.GetDirectoryName().Equals(package, StringComparison.OrdinalIgnoreCase))?.Path;
+        }
+
+        private static DirectoryPath GetPackagePath(DirectoryPath root, PackageReference package)
+        {
+            if (package.Parameters.ContainsKey("version"))
+            {
+                var version = package.Parameters["version"].First();
```

Comment:
This version may or may not be the "actual" version that is appended to the directory name when package is actually installed by `NuGet.Client´ libraries. The actual version comes from the package metadata and is always normalized to semver if possible. This is the main reason that the "old" `NuGetPackageInstaller` has the intermediate directory, as we could never be sure on what the actual install directory would end up as when using `nuget.exe`.